core/src/env.rs

```diff
@@ -144,6 +144,12 @@ pub trait Ext {
     /// Refund some gas.
     fn refund_gas(&mut self, amount: u64) -> Result<(), Self::Error>;
 
+    /// Reserve some gas for a few blocks.
+    fn reserve_gas(&mut self, amount: u32, blocks: u32) -> Result<ReservationId, Self::Error>;
+
+    /// Unreserve gas using reservation ID.
+    fn unreserve_gas(&mut self, id: ReservationId) -> Result<(), Self::Error>;
```

Comment:
What do you think about returning amount of gas being unreserved? It will probably be useful for userspace.

Reply:
- `gr_unreserve_gas` must be forbidden during gas calculation as `gr_gas_available` then
- We won't how much gas would be unreserved until program execution is done because we want to return leftovers of gas